rafx-assets/src/assets/asset_manager_render_resource.rs

```diff
@@ -1,23 +1,46 @@
-//TODO: Add a render resource builder or something that lets someone add extract-only things by
-// ref and resources available to all jobs by passing ownership
-
 use super::AssetManager;
 use rafx_base::memory::force_to_static_lifetime;
 use std::ops::Deref;
+use std::sync::Arc;
 
-// static reference is dangerous, must only be used when extracting
-pub struct AssetManagerRenderResource(&'static AssetManager);
+/// static reference is dangerous, must only be used when extracting. This is an option and is unset
+/// while not extracting.
+#[derive(Default)]
+pub struct AssetManagerRenderResource(Option<Arc<&'static AssetManager>>);
 
 impl AssetManagerRenderResource {
-    pub unsafe fn new(world: &AssetManager) -> Self {
-        AssetManagerRenderResource(force_to_static_lifetime(world))
+    /// Sets the AssetManager ref
+    pub unsafe fn begin_extract(
+        &mut self,
+        asset_manager: &AssetManager,
+    ) {
+        assert!(self.0.is_none());
+        self.0 = Some(Arc::new(force_to_static_lifetime(asset_manager)));
+    }
+
+    /// Clears the AssetManager ref and panics if any extract ref is remaining
+    pub fn end_extract(&mut self) {
+        self.0
+            .take()
+            .to_owned()
+            .expect("Reference to AssetManager is still in use");
+    }
+
+    /// Returns an extract ref. Panics if called while not extracting. The ref must be dropped
+    /// before end_extract() is called.
+    pub fn extract_ref(&self) -> AssetManagerExtractRef {
```

Comment:
Maybe rename this `get` to match swapchain? And we could have `try_get` that would return an option.

Reply:
I'm not sure about using get() for something that is returning anything more than a reference to a wrapped thing